src/main/kotlin/org/opensearch/replication/task/shard/ShardReplicationTask.kt

```diff
@@ -204,14 +204,9 @@ class ShardReplicationTask(id: Long, type: String, action: String, description:
         updateTaskState(FollowingState)
         val followerIndexService = indicesService.indexServiceSafe(followerShardId.index)
         val indexShard = followerIndexService.getShard(followerShardId.id)
-        // Adding retention lease at lastSyncedGlobalCheckpoint. This makes sure
-        // new tasks spawned after node changes/shard movements are handled properly
-        try {
-            logInfo("Adding retention lease")
-            retentionLeaseHelper.addRetentionLease(leaderShardId, indexShard.lastSyncedGlobalCheckpoint, followerShardId)
-        } catch (e: RetentionLeaseInvalidRetainingSeqNoException) {
-            logInfo("Retention lease add failed. Ignoring ${e.stackTraceToString()}")
-        }
+
+        retentionLeaseHelper.renewRetentionLease(leaderShardId, indexShard.lastSyncedGlobalCheckpoint, followerShardId)
+
```

Comment:
It looks like we introduced this explicitly to handle reroutes etc where Shard task cleans up retention lease while cancelling - is that not the case anymore?

https://github.com/opensearch-project/cross-cluster-replication/pull/138/files

Reply:
Looking into code, lease removal is happening in following code points
1. ShardReplicationTask - during task cleanup while it isn't moving to paused state
2.IndexReplicationTask - if replication is stopped in bootstrapping state
3. During resume API while performing validity for resume if one of the shard's retention lease doesn't exist
4. During Stop API if it was paused

If we can remove the cleanup from ShardReplicationStop and remove the "pause" check in Stop API in addition to change introduced in this PR should ensure all cases are appropriately handled.

Let me know your thoughts.

Reply:
I agree. we shouldn't be removing lease in the `ShardReplicationTask` on task cancellation. 

Currently I'm just checking why lease renewal was added in Stop API just for paused state. I've kept it as it is for now as i don't see any side effect.